### ISSUE TITLE
Information is missing when installing programs using Pop Shop.

### ISSUE THREAD

**Issue by Loughty** [pop-os/pop#3219]
**Distribution (run `cat /etc/os-release`):**
 
 Pop!_OS 22.04 LTS
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

```
pop-shop:
  Instalados: 7.3.0pop0~1695364353~22.04~e80e0b1
  Candidato:  7.3.0pop0~1695364353~22.04~e80e0b1
  Tabla de versión:
 *** 7.3.0pop0~1695364353~22.04~e80e0b1 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**

There is no progress bar when you install a new program, only status messages that are often insufficient. It also doesn't show when a program is queued.

**Steps to reproduce (if you know):**

Install a new program using Pop Shop.

(generic installation message, I don't know what it is installing at that moment or how much it takes to finish)
![Captura desde 2024-01-31 07-55-40](https://github.com/pop-os/pop/assets/52808915/9072b909-775e-42c1-9822-55a3a3e83979)

(program in queue, it only shows the cancel button, so I don't know if it is installing now or still in queue, or when it is its turn to install)
![Captura desde 2024-01-31 07-59-32](https://github.com/pop-os/pop/assets/52808915/d1341443-1ed3-4d7e-86c9-30a9b0915234)

**Expected behavior:**

- It should show a progress bar indicating how much has been downloaded.
- The status indicator should show, apart from the "installing" message, the name of the program that is currently being installed.
- If you are installing multiple programs, it should show when a program is queued.
- Programs that are being installed should be displayed in the programs list, or in a list dedicated to the installation queue.

**Other Notes:**

I'm not sure if this qualifies as a bug, but it's something I'd normally expect from a software store, and I found it extremely strange that Pop Shop didn't have it, even APT has a progress bar when you install something, and it shows you that it's installing at that moment, there is no need to hide it from the user.

Thanks in advance, and sorry if this is not the right place 🙏

**Comment by filipebsantos** [comment]
Yeah, this's really annoying. We can't see even the queue, simply none information.

**Comment by brys0** [comment]
Especially when there's over 10gb plus of updates. A progress bar and maybe even progress for each individual package would be highly beneficial. I understand the limitation when it comes to the program actually "installing" but a downloading percentage, or bar is far better than being left in the dark where the installer currently is. Or what is being downloaded/installed.

**Comment by Bo7a** [comment]
please consider this. If nothing else just surface the terminal output as an option, even if it is 100% non-interactive, just an echo of what is going on. I'm trying to be less of an old curmudgeon with pop-os and using GUI tools instead of always doing what I know at the CLI. But this 'up to 6.1GB" and "installing updates" is just not enough information.

**Comment by Joshfindit** [comment]
There are many many valid reasons for the user to have a long install:

1. A multi-GB install
2. Slow internet (possibly including packet loss that the user cannot control)
3. Slow drives
4. An old CPU
5. Some type of underlying hardware issue causing system slowdowns

And it's normal/expected for users to fall in to one of three categories:

1. impatient: "I want this right now, what's taking so long"
2. Multitasking: "If this is going to take a while I'll just go do something else while it works"
3. Concerned: "Is this still working?/Is something broken right now?"

In all cases the lack of any kind of progress is a poor experience.

*Worth noting: It "adds insult to injury" when the "Installing..." modal plays hide and seek with the cursor.*

**Comment by L0Lock** [comment]
when on the search result page you do see a progress bar:  
![image](https://github.com/pop-os/pop/assets/16049822/5adb1af6-953e-44ac-8818-64939b1729b7)

But it's a far cry from an actual page that shows the progress of all the ongoing tasks, their state and completion status, and sometimes even individual + global ETAs
